src/main/java/bio/terra/service/dataset/DatasetDao.java

```
@@ -93,16 +93,17 @@ public void lock(UUID datasetId, String flightId) {
     }
 
     /**
-     * Unlock the dataset object when finished doing something with it (e.g. delete, bulk file load).
-     * If the dataset is not locked by this flight, then the method is a no-op. It does not throw an exception in this
-     * case. So, multiple unlocks can succeed with no errors. The method does return a boolean indicating whether any
-     * rows were updated or not. So, callers can decide to throw an error if the unlock was a no-op.
+     * Release the exclusive lock on the dataset object when finished doing something with it (e.g. delete, create).
+     * If the dataset is not exclusively locked by this flight, then the method is a no-op. It does not throw an
+     * exception in this case. So, multiple unlocks can succeed with no errors. The method does return a boolean
```

Is there any time when you would want an exclusive lock to cause an exception?

---

I considered throwing two exceptions in unlock: DatasetNotFoundException if the id is invalid, DatasetLockException if the dataset is locked by a different flightid. Both cases would indicate a very serious bug (e.g. flight is coded wrong, we're getting a null for the flightid), and I think we'd have much bigger problems if those things were happening.

In the more common case of a step re-running an undo step, for example, I think failing silently is the preferred behavior.